test/upgrade/dry_run_does_not_apply_changes_test.dart

```diff
@@ -50,6 +50,50 @@ void main() {
     ]).validate();
   });
 
+  test('--dry-run: shows count of discontinued packages', () async {
```

Comment:
```suggestion
  test('Shows count of discontinued packages', () async {
```

Reply:
I think this test belongs better in for example test/upgrade/report/describes_change_test.dart . It does not really relate to dry-run.